TKeazirian.HTTPServer/Router/Router.cs

```diff
@@ -32,14 +33,19 @@ public Response Route(Request request)
 
         if (IsOptionsRequest(request))
         {
-            return new OptionsResponse(route).BuildOptionsResponse();
+            return new OptionsHandler(allowedMethods).HandleResponse(request);
         }
 
-        if (!route.MethodExistsForPath(method))
+        if (!IsMethodInHttpMethodsEnum(method))
         {
             return new NotImplementedResponse().BuildNotImplementedResponse();
         }
 
+        if (IsMethodInHttpMethodsEnum(method) && !route.MethodExistsForPath(method))
```

Comment:
Nitpick: I think referencing data types in function and variable names is a bit of a smell. What is `IsMethodInHttpMethodsEnum` really telling us? Whether it's a valid HTTP method?

Reply:
Good call...it's telling us whether the server knows the request method (aka it's in the enum), and then the second part of the logic tells us (if) it's not set up to support it.

Reply:
I thought `isValidHttpMethod` might work, but technically it could be a valid method that comes in, but this server is not set up to support it?  Because that is my understanding of what 405 means according to the MDN [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405).  501 ([docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501)) and 405 are a bit confusing.  405 indicates the server DOES recognize the method (to me, that would mean it's in the Enum) and 501 would mean it's not in the Enum.  

For the purposes of this without getting too bogged down in the definition of "valid", I still think `isValidHttpMethod` might work?

Reply:
Ah, yeah I see why that's tricky. I noticed that you're already checking `IsMethodInHttpMethodsEnum` on line 39, so is the additional check necessary here? Can we assume that it's `true` since we've passed the guard? If that's the case, then maybe what this function is really asking is "is the method implemented?"

Reply:
Oh interesting, yes that makes sense...And it reads well because if the method is called `IsHttpMethodImplemented` --> if false, then return `NotImplementedHandler`.